subsys/net/lib/download_client/src/download_client.c

```diff
@@ -433,7 +433,9 @@ static size_t socket_recv(struct download_client *dl)
 	case IPPROTO_DTLS_1_2:
 		if (IS_ENABLED(CONFIG_COAP)) {
 			timeout = coap_get_recv_timeout(dl);
-			if (timeout <= 0) {
+			if (timeout < 0) {
+				return -1;
```

Comment:
Should this set a custom errno then? Otherwise the behaviour is unpredictable.

Reply:
thx, i think an assert is the better choice here